src/components/layout/Vending/VendingForm.js

```diff
@@ -1,33 +1,118 @@
-import React, { useRef, useContext } from 'react';
+import React, {
+  useRef,
+  useContext,
+  useCallback,
+  useState,
+  useEffect,
+} from 'react';
+import styled from 'styled-components';
 import Input from '../../UI/Input';
+import Button from '../../UI/Button';
 import AmountContext from '../../../store/AmountContext';
 
-const VendingForm = (props) => {
+const checkValid = (wallet, money) => {
+  const newInsertedMoney = {};
+  const walletObj = Object.entries(wallet);
+  for (let i = walletObj.length - 1; i >= 0; i--) {
+    const [unit, count] = walletObj[i];
+    if (Number(unit) > money) continue;
+    if (Number(unit) * Number(count) < money) continue;
+    else {
+      let use = Math.floor(money / Number(unit));
+      money -= Number(unit) * use;
+      newInsertedMoney[unit] = use;
+    }
+  }
+
+  const moneyisValid = !money ? true : false;
+  return [moneyisValid, newInsertedMoney];
+};
+
+const VendingForm = () => {
   const amountInputRef = useRef();
-  const amountCtx = useContext(AmountContext);
+  const { money, dispatchMoney, dispatchLog } = useContext(AmountContext);
+  const [isInputValid, setIsInputValid] = useState(true);
+
   const submitHandler = (event) => {
     event.preventDefault();
-    amountCtx.add(amountInputRef.current.value);
+    const insertedMoney = amountInputRef.current.value;
+    const [isValid, newInsertedMoneyObj] = checkValid(
+      money.WALLET,
+      insertedMoney
+    );
+
+    if (isValid) {
+      setIsInputValid(true);
+      dispatchMoney({
+        type: 'INSERT',
+        newState: newInsertedMoneyObj,
```

Comment:
```suggestion
        payload: newInsertedMoneyObj,
```

보통 payload라는 이름을 사용하는 편이에요.